### DISCUSSION TITLE
First Release

### DISCUSSION THREAD

**Discussion by pschatzmann** [pschatzmann/arduino-audio-driver#1] [Announcements]
I was testing the functionality with an es8388 based AudioKit and the audio output if working perfectly

<hr /><em>This discussion was created from the release <a href='https://github.com/pschatzmann/arduino-audio-driver/releases/tag/v0.0.1'>First Release</a>.</em>

**Comment by robert1x2** [comment]
Hi, it's working like a charm. I use it with an ESP32 Audio Kit V2.2 A252 I recently bought. Its marked with ESP32-A1S. I use "ESP32 Dev Module" as Board in Arduino IDE. This I wasn't sure to use and took me while to find.

So I am using this code but don't really know how to bring the SD card / file send data to the copier. Any help appriciated.

```
#include "AudioTools.h"
#include "AudioCodecs/CodecMP3Helix.h"
#include "AudioLibs/AudioKit.h"

#define PIN_AUDIO_KIT_SD_CARD_CS 13
#define PIN_AUDIO_KIT_SD_CARD_MISO 2
#define PIN_AUDIO_KIT_SD_CARD_MOSI 15
#define PIN_AUDIO_KIT_SD_CARD_CLK  14

URLStream url("mywifiname","mywifipassword");  // or replace with ICYStream to get metadata
AudioKitStream i2s; // finale Ausgabe des decodierten Stream per i2s
EncodedAudioStream dec(&i2s, new MP3DecoderHelix()); // Decoding stream
StreamCopy copier(dec, url); // copy url to decoder


void setup(){
  Serial.begin(115200);

  SPI.begin(PIN_AUDIO_KIT_SD_CARD_CLK, PIN_AUDIO_KIT_SD_CARD_MISO, PIN_AUDIO_KIT_SD_CARD_MOSI, PIN_AUDIO_KIT_SD_CARD_CS);

  AudioLogger::instance().begin(Serial, AudioLogger::Info);  // nur zum Testen diese Zeile Aktivieren, um Details im Serial Monitor zu sehen
// AudioLogger::instance().begin(Serial, AudioLogger::Warning);  // Im Betrieb diese Zeile aktivieren

  

  // setup i2s
  auto config = i2s.defaultConfig(TX_MODE);
  i2s.begin(config);

  // setup I2S based on sampling rate provided by decoder
  dec.begin();

  // mp3 radio
  url.begin("http://stream.srg-ssr.ch/m/rsj/mp3_128","audio/mp3"); //Eintragen deines Wunschsenders

}

void loop(){
  copier.copy();
}
```

**Comment by pschatzmann** [comment]
With AudioKitStream you are still using the retired API.
I suggest that you have a look at the [updated examples](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit). There you can also see how to use the different SD libraries...
